### ISSUE TITLE
Improve implication endpoint in booster

### ISSUE THREAD

**Issue by goodlyrottenapple** [runtimeverification/haskell-backend#3854]

**Comment by ehildenb** [comment]
First we should measure what the effect is. Over the KEVM test-suite, for all the implications we do, how many and which ones fall back? What about for Kontrol? And for Kasmer?

We shouldn't speculatively implement _any_ reasoning, it should all be motivated by data.

**Comment by goodlyrottenapple** [comment]
Yes, this is why we didn't include it in the initial PR, the current implementation seems to be able to handle at least 80% of the cases in KEVM and Kontrol. Just putting this here to track the (in)completeness of the current booster implementation.
